woocommerce/google-listings-and-ads
# Add Merchant Center phone number controller

### Changes proposed in this Pull Request:

Part of #863.

This adds the `GET` and `POST` endpoints for retrieving and updating a Merchant Center account's phone number.

```json
# Retrieve
GET /wc/gla/mc/phone-number
# Response (200 or 4xx with details)
{
    "id": "123456789",
    "phone_number": "8675309"
}
```
```json
# Update
POST /wc/gla/mc/phone-number
# Body
{
    "phone_number": "(867) - 5309"
}
# Response (200 or 4xx with details)
{
    "id": "123456789",
    "phone_number": "8675309"
}
```

### Notes
- In the Google API, there's no validation at all on the phone number value (any character or even empty value). In the REST endpoint, some validation is performed (only phone-number type characters are allowed: `()+-. 012345689`, and no empty values – this means there's no way to _remove_ the phone number from the Merchant Center, [which isn't possible in the MC UI either)](https://user-images.githubusercontent.com/228780/127187002-5155f49d-9a8e-429a-a720-4724d12894f9.png).
- The number itself is then cleaned and left with only `+` and `0-9` values when sent to the Shopping API.
- An account with no phone number will have a `null` value.
- There's also a `verification_code` attribute for the `POST` request, which can be used later on when API verification is enabled.
- ~Not sure if it makes more sense to broaden the scope and allow full account details retrieval and updates to _any_ property of the account (with limits in the code) for future requirements.~ This will shortly be modified to be a _Business Information_ REST endpoint, for handling phone number, address, and phone verification.

### Detailed test instructions:

1. Connect a Merchant Center account.
2. Make a request to the `GET /wc/gla/mc/phone-number` endpoint
    - Confirm that no phone number is returned
3. Update the phone number using the `POST /wc/gla/mc/phone-number` endpoint
   - Confirm that the phone number is updated using the above `GET` endpoint
   - Confirm that the phone number is updated in Merchant Center : 
   <https://merchants.google.com/mc/merchantprofile/businessinfo>

(No changelog until the full feature is added).

### Changelog entry
